internal/rollup/rollup_bundle.bzl

```diff
@@ -48,47 +53,63 @@ def write_rollup_config(ctx, plugins=[]):
       output = config,
       template =  ctx.file._rollup_config_tmpl,
       substitutions = {
+          "TMPL_inputs": entry_points,
           "TMPL_bin_dir_path": ctx.bin_dir.path,
           "TMPL_workspace_name": ctx.workspace_name,
-          "TMPL_build_file_dirname": buildFileDirname,
+          "TMPL_build_file_dirname": build_file_dirname,
           "TMPL_label_name": ctx.label.name,
           "TMPL_module_mappings": str(mappings),
           "TMPL_additional_plugins": ",\n".join(plugins),
       })
 
   return config
 
-def run_rollup(ctx, config, output):
-  entryPoint = "/".join([ctx.workspace_name, ctx.attr.entry_point])
-
+def run_rollup(ctx, config):
   args = ctx.actions.args()
   args.add(["--config", config.path])
-  args.add(["--output.file", output.path])
-  args.add(["--input", entryPoint])
 
   es6_sources = collect_es6_sources(ctx)
 
+  output_dir = ctx.actions.declare_directory("bundles.es6")
+
   ctx.action(
       executable = ctx.executable._rollup,
       inputs = es6_sources + ctx.files.node_modules + [config],
-      outputs = [output],
+      outputs = [output_dir],
       arguments = [args]
   )
 
-def run_tsc(ctx, input, output):
+  return output_dir
+
+def run_tsc(ctx, input_dir):
+  config = ctx.actions.declare_file("_%s.tsconfig.json" % ctx.label.name)
+
+  output_dir = ctx.actions.declare_directory("bundles")
+
+  build_file_dirname = ctx.build_file_path.split("/")[:-1]
+  input_dir_rerooted = "/".join(input_dir.short_path.split("/")[len(build_file_dirname):])
+  output_dir_rerooted = "/".join(output_dir.short_path.split("/")[len(build_file_dirname):])
+
+  ctx.actions.expand_template(
+      output = config,
+      template =  ctx.file._tsconfig_tmpl,
+      substitutions = {
+          "TMPL_include_path": input_dir_rerooted + "/*",
+          "TMPL_out_dir": output_dir_rerooted,
+      })
+
   args = ctx.actions.args()
-  args.add(["--target", "es5"])
-  args.add("--allowJS")
-  args.add(input.path)
-  args.add(["--outFile", output.path])
+  args.add(["--project", config.path])
```

Comment:
tsconfig needed for tsc as its the only way to tell it to compile all files in a given folder. in this case the output folder of the rollup rule